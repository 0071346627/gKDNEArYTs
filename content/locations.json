{
  "LOCATIONS": [
    {
      "name": "london",
      "country": "united kingdom",
      "key": "london",
      "path": "/weather/london",
      "longitude": "51.5074",
      "latitude": "0.1278"
    },
    {
      "name": "paris",
      "country": "france",
      "key": "paris",
      "path": "/weather/paris",
      "longitude": "48.8566",
      "latitude": "2.3522"
    },
    {
      "name": "new york",
      "country": "united states",
      "key": "new-york",
      "path": "/weather/new-york",
      "longitude": "42.3601",
      "latitude": "-71.0589"
    },
    {
      "name": "singapore",
      "country": "singapore",
      "key": "singapore",
      "path": "/weather/singapore",
      "longitude": "1.3521",
      "latitude": "103.8198"
    },
    {
      "name": "sydney",
      "country": "australia",
      "key": "sydney",
      "path": "/weather/sydney",
      "longitude": "-33.8688",
      "latitude": "151.2093"
    }
  ]
}
