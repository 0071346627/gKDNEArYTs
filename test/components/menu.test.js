import Menu from '~/components/menu'
import { mount } from '@vue/test-utils'
import { LOCATIONS } from '~/content/locations.json'

const ACTIVE_PATH = '/weather/paris'
const propsData = { links: LOCATIONS, title: 'My title' }

describe('Menu', () => {
  let wrapper
  let html

  beforeAll(() => {
    wrapper = mount(Menu, {
      propsData,
      mocks: {
        $route: {
          path: ACTIVE_PATH
        }
      },
      stubs: ['nuxt-link']
    })
    html = wrapper.html()
  })

  test('html snapshot', () => {
    expect(html).toMatchSnapshot()
  })

  test('menu contains link tag for each provided link', () => {
    const tag = 'nuxt-link-stub'
    LOCATIONS.forEach(({ name, path }) => {
      expect(html).toContain(`<${tag} to="${path}">${name}</${tag}>`)
    })
  })

  test('menu label shows correct title', () => {
    expect(wrapper.find('.menu-label').text()).toMatch(propsData.title)
  })
})
