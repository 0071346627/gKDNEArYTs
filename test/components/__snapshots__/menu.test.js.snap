// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu html snapshot 1`] = `
"<aside class=\\"menu is-capitalized\\"><p class=\\"menu-label has-text-black\\">
    My title
  </p> <ul class=\\"menu-list\\"><li><nuxt-link-stub to=\\"/weather/london\\">london</nuxt-link-stub></li><li><nuxt-link-stub to=\\"/weather/paris\\">paris</nuxt-link-stub></li><li><nuxt-link-stub to=\\"/weather/new-york\\">new york</nuxt-link-stub></li><li><nuxt-link-stub to=\\"/weather/singapore\\">singapore</nuxt-link-stub></li><li><nuxt-link-stub to=\\"/weather/sydney\\">sydney</nuxt-link-stub></li></ul></aside>"
`;
